egui_plot/src/lib.rs

```diff
@@ -728,20 +728,20 @@ impl<'a> Plot<'a> {
     }
 
     /// Interact with and add items to the plot and finally draw it.
-    pub fn show<R>(
+    pub fn show<'b, R>(
         self,
         ui: &mut Ui,
-        build_fn: impl FnOnce(&mut PlotUi) -> R + 'a,
+        build_fn: impl FnOnce(&mut PlotUi<'b>) -> R + 'a,
```

Comment:
This was a tricky one.

The following example helped me with figuring out the correct lifetime annotation:

```rust
trait PlotItem: std::fmt::Debug {}

#[derive(Debug)]
struct Line<'a>(&'a [f64]);

impl<'a> PlotItem for Line<'a> {}

#[derive(Debug)]
struct PlotUi<'a>(Vec<Box<dyn PlotItem + 'a>>);

impl<'a> PlotUi<'a> {
    fn line(&mut self, line: Line<'a>) {
        self.0.push(Box::new(line));
    }
}

fn show<'a>(f: impl FnOnce(&mut PlotUi<'a>)) {
    let mut foo = PlotUi(Vec::new());
    f(&mut foo);
}

fn main() {
    let v = vec![1.0, 2.0];
    show(|plot_ui| {
        plot_ui.line(Line(&v));
        println!("{plot_ui:?}");
    });
}
```

Reply:
Sounds like we should add that code as an example or test then 🤔